spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/RetryGatewayFilterFactory.java

```diff
@@ -102,15 +102,17 @@ public GatewayFilter apply(RetryConfig retryConfig) {
 					return false;
 				}
 
-				for (Class<? extends Throwable> clazz : retryConfig.getExceptions()) {
-					if (clazz.isInstance(context.exception())) {
-						trace("exception is retryable %s, configured exceptions",
-								context.exception().getClass().getName(), retryConfig.getExceptions());
+				Throwable exception = context.exception();
+				for (Class<? extends Throwable> retryableClass : retryConfig.getExceptions()) {
+					if (retryableClass.isInstance(exception)
+							|| (exception != null && retryableClass.isInstance(exception.getCause()))) {
+						trace("exception is retryable %s, configured exceptions %s",
```

Comment:
should we log cause here instead if we are retries by cause? also does it make sense to add configuration key traverseCauses?

Reply:
No, I think this is sufficient.